distribution/src/bin/elasticsearch-keystore.bat

```diff
@@ -4,9 +4,8 @@ setlocal enabledelayedexpansion
 setlocal enableextensions
 
 set ES_MAIN_CLASS=org.elasticsearch.common.settings.KeyStoreCli
-call "%~dp0elasticsearch-cli.bat" ^
-  %%* ^
-  || exit /b 1
+call "%~dp0elasticsearch-cli.bat" %%* 
```

Comment:
Is there a reason this wouldn't work? Then we can keep the exit close to the failing script, we really do want to error out immediately here.

```suggestion
call "%~dp0elasticsearch-cli.bat" ^
  %%* ^
  || exit /b %errorlevel%
```

Reply:
Thanks for the review @jasontedor, finally took the effort to get back to this PR.

Sadly we need the `exit /b %errorlevel%` after the `endlocal` calls otherwise it does not survive in a non `CMD` context.

I also tried the special `:eof` label as [documented here](https://blogs.msdn.microsoft.com/oldnewthing/20120802-00/?p=6973)


```DOS
call "%~dp0elasticsearch-cli.bat" %%* || goto :eof
```

We could get rid of the `endlocal` calls alltogether since the local context is automatically closed when the bat file exits but non CMD contexts still need the explicit `exit /B`

I will update the PR with the following:

```bat
call "%~dp0elasticsearch-cli.bat" %%* || goto exit

endlocal
endlocal
:exit
exit /B %ERRORLEVEL%
```

Which I think makes the early exit explicit and satisfies carrying over the exit code to non CMD contexts.